packages/expo-image/src/web/AnimationManager.tsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+
+type AnimationManagerNode = [
+  key: string,
+  renderFunction: (callbacks: {
+    onReady: () => void;
+    onAnimationFinished: () => void;
+    onMount: () => void;
+    ref: React.RefObject<HTMLImageElement>;
+  }) => React.ReactElement
+];
+
+type Animation = null | {
+  run: (to: React.RefObject<HTMLImageElement>, from: React.RefObject<HTMLImageElement>[]) => void;
+  startingClass: string;
+  containerClass: string;
+  timingFunction: string | null;
+  animationClass: string | null;
+};
+
+function setClassOnElement(element: HTMLImageElement | null, classes: string[]) {
+  if (!element) {
+    return;
+  }
+  element.setAttribute('class', classes.join(' '));
+}
+
+function useAnimationManagerNode(node: AnimationManagerNode | null) {
+  const callbackContainer: {
+    onReady: (() => void) | null;
+    onAnimationFinished: (() => void) | null;
+    onMount: (() => void) | null;
+  } = {
+    onReady: null,
+    onAnimationFinished: null,
+    onMount: null,
+  };
+  const newNode = React.useMemo(() => {
+    if (!node) {
+      return null;
+    }
+    const [animationKey, renderFunction] = node;
+    const ref = React.createRef<HTMLImageElement>();
+    const child = renderFunction({
+      onReady: () => {
+        callbackContainer.onReady?.();
+      },
+      onAnimationFinished: () => {
+        callbackContainer.onAnimationFinished?.();
+      },
+      onMount: () => {
+        callbackContainer.onMount?.();
+      },
+      ref,
+    });
+    // key, ReactElement, ref, callbacks
+    return { animationKey, child, ref, callbackContainer };
+  }, [node?.[1]]);
+  return newNode;
+}
+
+const validateTimingFunctionForAnimation = (
+  animationClass: string | null,
+  timingFunction: string | null
+) => {
```

Comment:
```suggestion
function validateTimingFunctionForAnimation(
  animationClass: string | null,
  timingFunction: string | null
) {
```

I guess the `timingFunction` could be of `ImageTransition['timing']` type